ClassTag on Chunk is having issue in ScalaJS

while working on #3358, found a bug on the master.

To reproduce : 
```scala
zio.test.test("0 to 128") {
      (1 to 128).foldLeft(Chunk(1))(_ + _).materialize
      assertCompletes
    }
```
The error:
```scala
[info] ZIO Test
[info] - ChunkSpec
[info]   - 0 to 128
[info]     Fiber failed.
[info]     An unchecked error was produced.
[info]     scala.scalajs.runtime.UndefinedBehaviorError: An undefined behavior was detected: 128 is not an instance of java.lang.Byte
[info]       at scala.scalajs.runtime.UndefinedBehaviorError.fillInStackTrace(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:49052:14)
[info]       at scala.scalajs.runtime.UndefinedBehaviorError.fillInStackTrace(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:197175:67)
[info]       at scala.scalajs.runtime.UndefinedBehaviorError.<init>(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:49176:10)
[info]       at scala.scalajs.runtime.UndefinedBehaviorError.<init>(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:197182:58)
[info]       at scala.scalajs.runtime.UndefinedBehaviorError.<init>(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:197178:67)
[info]       at $throwClassCastException(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:217:46)
[info]       at $asByte(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:650:5)
[info]       at $uB(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:689:10)
[info]       at scala.runtime.ScalaRunTime$.array$undupdate(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:33744:17)
[info]       at zio.Chunk$Singleton.toArray(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:222574:25)
[info]     Caused by: java.lang.ClassCastException: 128 is not an instance of java.lang.Byte
[info]       at java.lang.ClassCastException.fillInStackTrace(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:49052:14)
[info]       at java.lang.ClassCastException.<init>(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:49176:10)
[info]       at java.lang.ClassCastException.<init>(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:171184:58)
[info]       at $throwClassCastException(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:218:36)
[info]       at $asByte(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:650:5)
[info]       at $uB(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:689:10)
[info]       at scala.runtime.ScalaRunTime$.array$undupdate(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:33744:17)
[info]       at zio.Chunk$Singleton.toArray(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:222574:25)
[info]       at zio.Chunk$Concat.toArray(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:222205:12)
[info]       at zio.Chunk.toArray(/Users/jon/Projects/zio/core-tests/js/target/scala-2.12/core-tests-test-fastopt.js:216701:9)
[info]     
[info]     Fiber:Id(1586828934537,12975) was supposed to continue to: <empty trace>
[info]     
[info]     Fiber:Id(1586828934537,12975) ZIO Execution trace: <empty trace>
[info]     
[info]     Fiber:Id(1586828934537,12975) was spawned by: <empty trace>
[info] Done
[error] Failed: Total 64, Failed 1, Errors 0, Passed 63, Ignored 2
[error] Failed tests:
[error]         zio.ChunkSpec

```

using 
```
sbt> coreTestsJS/testOnly *.ChunkSpec
```

discovered the same issue for `zio.Chunk$PrependN.toArray` in #3914 on master
can be reproduced with https://github.com/zio/zio/pull/3914/commits/73f14a989b98d15369cd9e389f6453f253153133 with ScalaJs 2.13, jdk8 (see https://app.circleci.com/pipelines/github/zio/zio/8228/workflows/25f29f1e-ed92-46e4-9987-76113cf7d47a/jobs/83141/steps)

#3418 does not fix this one

```scala
        testM("preserves data")(checkM(Gen.chunkOf(Gen.anyInt.filter(_ != 0))) { ints =>
          val splitSequence: Chunk[Int] = Chunk(0, 1)
          val data: Chunk[Int]          = ints.flatMap((i: Int) => i +: splitSequence)
          val parser                    = ZTransducer.splitOnChunk[Int, Chunk[Int]](splitSequence, identity)
          assertM(run(parser, List(data)).map(_.flatten))(equalTo(ints))
        })
```

```
[info]         Fiber failed.
[info]         An unchecked error was produced.
[info]         org.scalajs.linker.runtime.UndefinedBehaviorError: java.lang.ClassCastException: -246096894 is not an instance of java.lang.Byte
[info]           at $c_jl_Throwable(/home/circleci/project/streams-tests/js/target/scala-2.13/streams-tests-test-fastopt.js:39949:5)
[info]           at $c_jl_Error(/home/circleci/project/streams-tests/js/target/scala-2.13/streams-tests-test-fastopt.js:63107:1)
[info]           at $c_jl_VirtualMachineError(/home/circleci/project/streams-tests/js/target/scala-2.13/streams-tests-test-fastopt.js:77047:1)
[info]           at $c_Lorg_scalajs_linker_runtime_UndefinedBehaviorError(/home/circleci/project/streams-tests/js/target/scala-2.13/streams-tests-test-fastopt.js:96837:5)
[info]           at $throwClassCastException(/home/circleci/project/streams-tests/js/target/scala-2.13/streams-tests-test-fastopt.js:27:9)
[info]           at $uB(/home/circleci/project/streams-tests/js/target/scala-2.13/streams-tests-test-fastopt.js:378:52)
[info]           at scala.runtime.ScalaRunTime$.array_update(/home/circleci/project/streams-tests/js/target/scala-2.13/streams-tests-test-fastopt.js:15165:19)
[info]           at scala.Array$.slowcopy(/home/circleci/project/streams-tests/js/target/scala-2.13/streams-tests-test-fastopt.js:48189:27)
[info]           at scala.Array$.copy(/home/circleci/project/streams-tests/js/target/scala-2.13/streams-tests-test-fastopt.js:48235:7)
[info]           at zio.Chunk$PrependN.toArray(/home/circleci/project/streams-tests/js/target/scala-2.13/streams-tests-test-fastopt.js:209348:19)
[info]         Caused by: java.lang.ClassCastException: -246096894 is not an instance of java.lang.Byte
[info]           at $c_jl_Throwable(/home/circleci/project/streams-tests/js/target/scala-2.13/streams-tests-test-fastopt.js:39949:5)
[info]           at $c_jl_Exception(/home/circleci/project/streams-tests/js/target/scala-2.13/streams-tests-test-fastopt.js:63140:1)
[info]           at $c_jl_RuntimeException(/home/circleci/project/streams-tests/js/target/scala-2.13/streams-tests-test-fastopt.js:76701:1)
[info]           at $c_jl_ClassCastException(/home/circleci/project/streams-tests/js/target/scala-2.13/streams-tests-test-fastopt.js:94786:1)
[info]           at $throwClassCastException(/home/circleci/project/streams-tests/js/target/scala-2.13/streams-tests-test-fastopt.js:27:98)
[info]           ... 5 more
```

--- a/core-tests/shared/src/test/scala/zio/ChunkSpec.scala
+++ b/core-tests/shared/src/test/scala/zio/ChunkSpec.scala
@@ -526,6 +526,10 @@ object ChunkSpec extends ZIOBaseSpec {
               else IO.succeed(i)
           }
       )(equalTo(2))
+    },
+    zio.test.test("Tags.fromValue is safe on Scala.is") {
+      val _ = Chunk(1, 128)
+      assertCompletes
     }
   )
 }

--- a/core/js/src/main/scala/zio/ChunkPlatformSpecific.scala
+++ b/core/js/src/main/scala/zio/ChunkPlatformSpecific.scala
@@ -1,0 +1,13 @@
+package zio
+
+import scala.reflect.{ classTag, ClassTag }
+
+private[zio] trait ChunkPlatformSpecific {
+
+  private[zio] object Tags {
+    def fromValue[A](a: A): ClassTag[A] = {
+      val _ = a
+      classTag[AnyRef].asInstanceOf[ClassTag[A]]
+    }
+  }
+}

--- a/core/jvm/src/main/scala/zio/ChunkPlatformSpecific.scala
+++ b/core/jvm/src/main/scala/zio/ChunkPlatformSpecific.scala
@@ -1,0 +1,56 @@
+package zio
+
+import scala.reflect.{ classTag, ClassTag }
+
+private[zio] trait ChunkPlatformSpecific {
+
+  private[zio] object Tags {
+    def fromValue[A](a: A): ClassTag[A] =
+      unbox(ClassTag(a.getClass))
+
+    private def unbox[A](c: ClassTag[A]): ClassTag[A] =
+      if (isBoolean(c)) BooleanClass.asInstanceOf[ClassTag[A]]
+      else if (isByte(c)) ByteClass.asInstanceOf[ClassTag[A]]
+      else if (isShort(c)) ShortClass.asInstanceOf[ClassTag[A]]
+      else if (isInt(c)) IntClass.asInstanceOf[ClassTag[A]]
+      else if (isLong(c)) LongClass.asInstanceOf[ClassTag[A]]
+      else if (isFloat(c)) FloatClass.asInstanceOf[ClassTag[A]]
+      else if (isDouble(c)) DoubleClass.asInstanceOf[ClassTag[A]]
+      else if (isChar(c)) CharClass.asInstanceOf[ClassTag[A]]
+      else classTag[AnyRef].asInstanceOf[ClassTag[A]] // TODO: Find a better way
+
+    private def isBoolean(c: ClassTag[_]): Boolean =
+      c == BooleanClass || c == BooleanClassBox
+    private def isByte(c: ClassTag[_]): Boolean =
+      c == ByteClass || c == ByteClassBox
+    private def isShort(c: ClassTag[_]): Boolean =
+      c == ShortClass || c == ShortClassBox
+    private def isInt(c: ClassTag[_]): Boolean =
+      c == IntClass || c == IntClassBox
+    private def isLong(c: ClassTag[_]): Boolean =
+      c == LongClass || c == LongClassBox
+    private def isFloat(c: ClassTag[_]): Boolean =
+      c == FloatClass || c == FloatClassBox
+    private def isDouble(c: ClassTag[_]): Boolean =
+      c == DoubleClass || c == DoubleClassBox
+    private def isChar(c: ClassTag[_]): Boolean =
+      c == CharClass || c == CharClassBox
+
+    private val BooleanClass    = classTag[Boolean]
+    private val BooleanClassBox = classTag[java.lang.Boolean]
+    private val ByteClass       = classTag[Byte]
+    private val ByteClassBox    = classTag[java.lang.Byte]
+    private val ShortClass      = classTag[Short]
+    private val ShortClassBox   = classTag[java.lang.Short]
+    private val IntClass        = classTag[Int]
+    private val IntClassBox     = classTag[java.lang.Integer]
+    private val LongClass       = classTag[Long]
+    private val LongClassBox    = classTag[java.lang.Long]
+    private val FloatClass      = classTag[Float]
+    private val FloatClassBox   = classTag[java.lang.Float]
+    private val DoubleClass     = classTag[Double]
+    private val DoubleClassBox  = classTag[java.lang.Double]
+    private val CharClass       = classTag[Char]
+    private val CharClassBox    = classTag[java.lang.Character]
+  }
+}

--- a/core/native/src/main/scala/zio/ChunkPlatformSpecific.scala
+++ b/core/native/src/main/scala/zio/ChunkPlatformSpecific.scala
@@ -1,0 +1,13 @@
+package zio
+
+import scala.reflect.{ classTag, ClassTag }
+
+private[zio] trait ChunkPlatformSpecific {
+
+  private[zio] object Tags {
+    def fromValue[A](a: A): ClassTag[A] = {
+      val _ = a
+      classTag[AnyRef].asInstanceOf[ClassTag[A]]
+    }
+  }
+}

--- a/core/shared/src/main/scala/zio/Chunk.scala
+++ b/core/shared/src/main/scala/zio/Chunk.scala
@@ -1067,7 +1067,7 @@ sealed trait Chunk[+A] extends ChunkLike[A] { self =>
     }
 }
 
-object Chunk extends ChunkFactory {
+object Chunk extends ChunkFactory with ChunkPlatformSpecific {
 
   /**
    * Returns the empty chunk.
@@ -1718,56 +1718,6 @@ object Chunk extends ChunkFactory {
 
     override private[zio] def reverseArrayIterator[A]: Iterator[Array[A]] =
       Iterator.empty
-  }
-
-  private[zio] object Tags {
-    def fromValue[A](a: A): ClassTag[A] =
-      unbox(ClassTag(a.getClass))
-
-    private def unbox[A](c: ClassTag[A]): ClassTag[A] =
-      if (isBoolean(c)) BooleanClass.asInstanceOf[ClassTag[A]]
-      else if (isByte(c)) ByteClass.asInstanceOf[ClassTag[A]]
-      else if (isShort(c)) ShortClass.asInstanceOf[ClassTag[A]]
-      else if (isInt(c)) IntClass.asInstanceOf[ClassTag[A]]
-      else if (isLong(c)) LongClass.asInstanceOf[ClassTag[A]]
-      else if (isFloat(c)) FloatClass.asInstanceOf[ClassTag[A]]
-      else if (isDouble(c)) DoubleClass.asInstanceOf[ClassTag[A]]
-      else if (isChar(c)) CharClass.asInstanceOf[ClassTag[A]]
-      else classTag[AnyRef].asInstanceOf[ClassTag[A]] // TODO: Find a better way
-
-    private def isBoolean(c: ClassTag[_]): Boolean =
-      c == BooleanClass || c == BooleanClassBox
-    private def isByte(c: ClassTag[_]): Boolean =
-      c == ByteClass || c == ByteClassBox
-    private def isShort(c: ClassTag[_]): Boolean =
-      c == ShortClass || c == ShortClassBox
-    private def isInt(c: ClassTag[_]): Boolean =
-      c == IntClass || c == IntClassBox
-    private def isLong(c: ClassTag[_]): Boolean =
-      c == LongClass || c == LongClassBox
-    private def isFloat(c: ClassTag[_]): Boolean =
-      c == FloatClass || c == FloatClassBox
-    private def isDouble(c: ClassTag[_]): Boolean =
-      c == DoubleClass || c == DoubleClassBox
-    private def isChar(c: ClassTag[_]): Boolean =
-      c == CharClass || c == CharClassBox
-
-    private val BooleanClass    = classTag[Boolean]
-    private val BooleanClassBox = classTag[java.lang.Boolean]
-    private val ByteClass       = classTag[Byte]
-    private val ByteClassBox    = classTag[java.lang.Byte]
-    private val ShortClass      = classTag[Short]
-    private val ShortClassBox   = classTag[java.lang.Short]
-    private val IntClass        = classTag[Int]
-    private val IntClassBox     = classTag[java.lang.Integer]
-    private val LongClass       = classTag[Long]
-    private val LongClassBox    = classTag[java.lang.Long]
-    private val FloatClass      = classTag[Float]
-    private val FloatClassBox   = classTag[java.lang.Float]
-    private val DoubleClass     = classTag[Double]
-    private val DoubleClassBox  = classTag[java.lang.Double]
-    private val CharClass       = classTag[Char]
-    private val CharClassBox    = classTag[java.lang.Character]
   }
 
   final case class AnyRefArray[A <: AnyRef](array: Array[A]) extends Arr[A]

